nemo/collections/asr/models/label_models.py

```diff
@@ -319,97 +318,56 @@ def training_step(self, batch, batch_idx):
         self._macro_accuracy.update(preds=logits, target=labels)
-        self._auroc.update(preds=logits, target=labels)
+        stats = self._macro_accuracy._get_final_stats()
 
         return {
-            'val_loss': loss_value,
-            'val_correct_counts': correct_counts,
-            'val_total_counts': total_counts,
-            'val_acc_top_k': acc_top_k,
+            f'{tag}_loss': loss_value,
+            f'{tag}_correct_counts': correct_counts,
+            f'{tag}_total_counts': total_counts,
+            f'{tag}_acc_micro_top_k': acc_top_k,
+            f'{tag}_acc_macro_stats': stats
         }
 
-    def multi_validation_epoch_end(self, outputs, dataloader_idx: int = 0):
-        val_loss_mean = torch.stack([x['val_loss'] for x in outputs]).mean()
-        correct_counts = torch.stack([x['val_correct_counts'] for x in outputs]).sum(axis=0)
-        total_counts = torch.stack([x['val_total_counts'] for x in outputs]).sum(axis=0)
+    def multi_validation_epoch_end(self, outputs, dataloader_idx: int = 0, tag: str = 'val'):
+        loss_mean = torch.stack([x[f'{tag}_loss'] for x in outputs]).mean()
+        correct_counts = torch.stack([x[f'{tag}_correct_counts'] for x in outputs]).sum(axis=0)
+        total_counts = torch.stack([x[f'{tag}_total_counts'] for x in outputs]).sum(axis=0)
 
         self._accuracy.correct_counts_k = correct_counts
         self._accuracy.total_counts_k = total_counts
         topk_scores = self._accuracy.compute()
+        
+        self._macro_accuracy.tp = torch.stack([x[f'{tag}_acc_stats'][0] for x in outputs]).sum(axis=0)
+        self._macro_accuracy.fp = torch.stack([x[f'{tag}_acc_stats'][1] for x in outputs]).sum(axis=0)
+        self._macro_accuracy.tn = torch.stack([x[f'{tag}_acc_stats'][2] for x in outputs]).sum(axis=0)
+        self._macro_accuracy.fn = torch.stack([x[f'{tag}_acc_stats'][3] for x in outputs]).sum(axis=0)
         macro_accuracy_score = self._macro_accuracy.compute()
-        auroc_score = self._auroc.compute()
 
         self._accuracy.reset()
         self._macro_accuracy.reset()
-        self._auroc.reset()
 
-        logging.info("val_loss: {:.3f}".format(val_loss_mean))
-        self.log('val_loss', val_loss_mean)
+        tensorboard_log = {
+            f'{tag}_loss': oss_mean,
+            f'{tag}_acc_macro_top@1': macro_accuracy_score,
+        }
+
         for top_k, score in zip(self._accuracy.top_k, topk_scores):
-            self.log('val_epoch_accuracy_top@{}'.format(top_k), score)
+            tensorboard_log[f'{tag}_acc_micro_top@{top_k}'] = score
 
-        self.log('val_acc_macro', macro_accuracy_score, sync_dist=True)
-        self.log('val_auroc', auroc_score, sync_dist=True)
-
-        return {
-            'val_loss': val_loss_mean,
-            'val_acc_top_k': topk_scores,
-            'val_acc_macro': macro_accuracy_score,
-            'val_auroc': auroc_score,
-        }
+        return {'log': tensorboard_log}
```

Comment:
also remove this tensorboard_logs, refer to https://github.com/NVIDIA/NeMo/pull/4876

Reply:
Sure, just fixed.